src/pybind/mgr/dashboard/frontend/e2e/cluster/osds.e2e-spec.ts

```diff
@@ -32,4 +33,51 @@ describe('OSDs page', () => {
       expect(osds.getTabText(1)).toEqual('Overall Performance');
     });
   });
+
+  describe('check existence of fields on OSD page', () => {
+    it('should check that number of rows and count in footer match', () => {
+      osds.navigateTo();
+      osds
+        .getTableCount()
+        .getText()
+        .then((tableCount) => {
+          tableCount = tableCount.slice(13, -6); // Grabs number of hosts from table footer
+          expect(osds.getTableRows().count()).toMatch(tableCount);
+        });
+    });
+
+    it('should verify that selected footer increases when an entry is clicked', () => {
+      osds.navigateTo();
+      osds.getFirstCell().click(); // clicks first osd
+      osds
+        .getTableCount()
+        .getText()
+        .then((tableCount) => {
+          tableCount = tableCount.substring(0, 1); // Grabs number of hosts from table footer
+          expect(tableCount).toMatch('1');
+        });
+    });
+
+    it('should verify that buttons exist', () => {
+      osds.navigateTo();
+      expect(element(by.cssContainingText('button', 'Scrub')).isPresent()).toBe(true);
+      expect(
+        element(by.cssContainingText('button', 'Cluster-wide configuration')).isPresent()
+      ).toBe(true);
+    });
+
+    it('should check the number of tabs when selecting an osd is correct', () => {
+      osds.navigateTo();
+      osds.getFirstCell().click(); // clicks first osd
+      expect(osds.getTabsCount()).toEqual(7); // includes tabs at the top of the page
+    });
+
+    it('should show the correct text for the tab labels', () => {
+      expect(osds.getTabText(2)).toEqual('Attributes (OSD map)');
+      expect(osds.getTabText(3)).toEqual('Metadata');
+      expect(osds.getTabText(4)).toEqual('Performance counter');
+      expect(osds.getTabText(5)).toEqual('Histogram');
+      expect(osds.getTabText(6)).toEqual('Performance Details');
```

Comment:
Could you please convert this e2e test into a unit tests inside `osd-list.component.spec.ts`?

Reply:
I assumed they were applicable based off of https://github.com/ceph/ceph/pull/25010

Reply:
This PR differs IMHO a lot from yours as it allowed protractor to navigate to the page and inspect errors on it while doing that. You are now extending an e2e test suite which already navigates to a page and watches for errors.

Furthermore there wasn't a tab text expectation.

As said before place them into unit tests suite or convert them into valid integration test. Found a good example yesterday in #29469

Reply:
I agree with @rafaelquint here. The code does not introduce a new type of tests, just extends the suite by tests of a kind that are already in the code.